### ISSUE TITLE
Default `string-quotes` rule and Sass modules

### ISSUE THREAD

**Issue by niksy** [stylelint/stylelint#5138]
> What is the problem you're trying to solve?

It seems like default `string-quotes` rule ignores Sass Modules at-rules `@use` and `@forward`. Is this something that should be supported in core? It seems there’s a [bunch of checks for non-standard configurations](https://github.com/stylelint/stylelint/blob/master/lib/rules/string-quotes/index.js) so I don’t know if that relates to Sass modules.

I’ve already submitted it to [stylelint-scss](https://github.com/kristerkari/stylelint-scss/issues/508), but author consesus is that it should be handled in core.

> What solution would you like to see?

Is it possible to add certain at-rules to allowed list of rules which should follow definition for this rule?

**Comment by jeddy3** [comment]
@niksy Thanks for the request and for using the template.

I can't reproduce this using [the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIACAzCAnA7gQ3VABADogoCuAdmAC4CWE5AzsQNyHmoY55EhUAWMALYxi+bA3z8hMALQAqVu1IMYPQdiox0NbABtRmGv3wAKACRThMgA7b16AJ7x8AI12kYASkVsQAGhAGB3IqbAAPBHAGBn9wehQaAHNI4DZ8HnRSXRgmBHxU8nT04gYqbXJEmQBHUghNXOcSmgrs4jT8AF82DpAOoA):

Given:

```scss
@forward "functions";
@forward "theme" as theme-*;
@use "material" with ($theme-primary: blue);
```

And:

```json
{
  "rules": {
    "string-quotes": "single"
  }
}
```

I get:

```console
1:10 error Expected single quotes (string-quotes)
2:10 error Expected single quotes (string-quotes)
3:6 error Expected single quotes (string-quotes)
```

The rules built into stylelint supports non-standard syntax _if they extend standard constructs_, like at-rules, functions and so on, which `@use` and `@forward` do. 

What behaviour are you seeing, and can you include a demo link to help us investigate?

**Comment by niksy** [comment]
I finally found out why this is happening.

I’ve been using Prettier for formatting the code, and currently there’s a [bug](https://github.com/prettier/prettier/issues/11053) where `use` and `forward` at-rules are not handled.

I’ve created [PR](https://github.com/prettier/prettier/pull/11461) on Prettier repo to fix this.
